components/kubeflow/pytorch-launcher/sample.py

```diff
@@ -0,0 +1,125 @@
+    return current_namespace
+
+
+pytorchjob_launcher_op = components.load_component_from_file("./component.yaml")
+# pytorchjob_launcher_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/master/components/kubeflow/pytorch-launcher/component.yaml')
+
+
+@dsl.pipeline(
+    name="Launch kubeflow pytorchjob",
+    description="An example to launch pytorch."
+)
+def mnist_train(
+        name: str="mnist",
+        namespace: str=get_current_namespace(),
+        worker_replicas: int=3,
+        ttl_seconds_after_finished: int=-1,
+        job_timeout_minutes: int=600,
+        delete_after_done: bool=False):
+    # Define the master and worker definitions by dict.
+    # These can also be defined using the kubeflow-pytorchjob API
+    master = {
+        "replicas": 1,
+        "restartPolicy": "OnFailure",
+        "template": {
+            "metadata": {
+                "annotations": {
+                    # See https://github.com/kubeflow/website/issues/2011
+                    "sidecar.istio.io/inject": "false"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                    # To override default command
+                    # "command": [
+                    #   "python",
+                    #   "/opt/mnist/src/mnist.py"
+                    # ],
+                    "args": [
+                        "--backend", "gloo",
+                    ],
+                    # Or, create your own image from 
+                    # https://github.com/kubeflow/pytorch-operator/tree/master/examples/mnist
+                    "image": "gcr.io/kubeflow-examples/pytorch-dist-mnist:v20180702-a57993c",
+                    "name": "pytorch",
+                    "resources": {
+                        "requests": {
+                            "memory": "4Gi",
+                            "cpu": "2000m",
+                        },
+                        "limits": {
+                            "memory": "4Gi",
+                            "cpu": "2000m",
+                        },
+                    },
+                    # If using GPUs
+                    # "resources": {
+                    #   "limits": {
+                    #     "nvidia.com/gpu": 1,
+                    #   }
+                    # }
+                    }
+                ],
+                # If imagePullSecrets required
+                # "imagePullSecrets": [
+                #     {"name": "image-pull-secret"},
+                # ],
+            },
+        },
+    }
+
+    # You could have a worker spec that is different from master
+    # (maybe master needs a PVC for saving results)
+    # This does not handle replicas==0.  Would need if statement at kfp
+    # runtime to handle that (doing 'if worker_replicas > 0' here does
+    # not work)
+    worker = copy.deepcopy(master)
+    worker['replicas'] = worker_replicas
```

Comment:
Can we make a component for constructing `worker` spec? 
It takes `work_replica: int` as an input (maybe more inputs if you need to parameterize anything else in the `worker` spec, and produces `worker: str` as an output which is then feed into the next component.

A pipeline is for organizing and connecting components. Ideally there shouldn't be any custom logics/operations sitting outside the components, arguably even it's as trivial as constructing strings. When you try to combine dynamic inputs in some way, you're dealing with the placeholders of the inputs, and this may incur unexpected behaviors like the one you mentioned below with `str()` vs. `json.dumps()`.

Reply:
I think this is a great idea and would follow what is done in the tfjob [sample](https://github.com/kubeflow/pipelines/blob/master/components/kubeflow/launcher/sample.py), although it could also be a separate task.  I believe @jagadeeshi2i will take a look as he continues this though